### ISSUE TITLE
Support for Python3.5 (Anaconda)

### ISSUE THREAD

**Issue by nowke** [Kozea/WeasyPrint#316]
I tried installing WeasyPrint through [Anaconda](https://anaconda.org/) using `conda` which gave error:

```bash
$ conda install -c jlmenut weasyprint=0.23
Using Anaconda Cloud api site https://api.anaconda.org
Fetching package metadata: ......
Solving package specifications: ..............
Error: Unsatisfiable package specifications.
Generating hint: 
[      COMPLETE      ]|###################################################| 100%

Hint: the following packages conflict with each other:
  - weasyprint 0.23*
  - python 3.5*

Use 'conda info weasyprint' etc. to see the dependencies for each package.
``` 

And also, using pip while Anaconda was set in `$PATH`

`pip install weasyprint`

Installation is successful, but the `weasyprint` is producing errors.

```bash
$ weasyprint http://www.w3.org/TR/CSS21/intro.html CSS21-intro.pdf -s \
 http://weasyprint.org/samples/CSS21-print.css
...
...
During handling of the above exception, another exception occurred:

SystemError: <cdata 'void(*)(PangoLayoutIter *)' 0x7f7203ee3280> returned a result with an error set
From callback for ffi.gc <cdata 'PangoLayoutIter *' 0x2fb5910>:
StopIteration
...
...
```

After, I disable `conda` by removing it from `$PATH` and install `weasyprint` through `pip`, it just works fine. Seems to be `Python3.5` issue.

**Comment by SimonSapin** [comment]
> ```
> Hint: the following packages conflict with each other:
>  - weasyprint 0.23*
>  - python 3.5*
> ```

I don’t know why they supposedly conflict, but I also don’t know what Anaconda is. Are they re-packaging stuff? Could this be a bug in their package?

> ```
> During handling of the above exception, another exception occurred:
> ```

"the above exception" could be relevant. Could you paste the full output?

**Comment by nowke** [comment]
The conflict occurs because of `Python 3.5`.
Checkout the similar issue [here](https://github.com/sunpy/sunpy/issues/1695) for `sunpy`

Regarding the exception, I'm pasting from the start

```bash
$ weasyprint http://www.w3.org/TR/CSS21/intro.html CSS21-intro.pdf -s \
http://weasyprint.org/samples/CSS21-print.css 

Ignored `columns: 12em` at 142:12, unknown property.
From callback for ffi.gc <cdata 'PangoLayoutIter *' 0x2984c90>:
StopIteration

During handling of the above exception, another exception occurred:

SystemError: <cdata 'void(*)(PangoLayoutIter *)' 0x7fce492e5280> returned a result with an error set
From callback for ffi.gc <cdata 'PangoLayout *' 0x29b8030>:
StopIteration
...
```

The Line from `SystemError` repeats until the last (upto 17k lines). It doesn't occur in Python2.7 or Python3.4. Nevertheless, _**PDF is generated**_ after giving 17k lines of errors.

**Comment by liZe** [comment]
It definitely looks like a bug in the way libffi, pango or cairo is packaged in Anaconda. As far as I can tell, it's not really related to WeasyPrint, and there's probably nothing we can do for you :sad:. @nowke could you try to update your packages and see if the bug is fixed?

**Comment by mofr** [comment]
I have successfully installed `weasyprint==0.30` into my anaconda using `pip install weasyprint`.
Unfortunately it fails with following traceback:
```
Traceback (most recent call last):
  File "hello.py", line 127, in <module>
    from weasyprint import HTML
  File "C:\Anaconda3\lib\site-packages\weasyprint\__init__.py", line 341, in <module>
    from .html import find_base_url, HTML5_UA_STYLESHEET, get_html_metadata  # noqa
  File "C:\Anaconda3\lib\site-packages\weasyprint\html.py", line 41, in <module>
    HTML5_UA_STYLESHEET = CSS(filename=os.path.join(root, 'css', 'html5_ua.css'))
  File "C:\Anaconda3\lib\site-packages\weasyprint\__init__.py", line 226, in __init__
    with result as (source_type, source, base_url, protocol_encoding):
  File "C:\Anaconda3\lib\contextlib.py", line 59, in __enter__
    return next(self.gen)
  File "C:\Anaconda3\lib\site-packages\weasyprint\__init__.py", line 298, in _select_source
    base_url = path2url(filename)
  File "C:\Anaconda3\lib\site-packages\weasyprint\urls.py", line 87, in path2url
    path = pathname2url(path)
  File "C:\Anaconda3\lib\nturl2path.py", line 48, in pathname2url
    if not ':' in p:
TypeError: a bytes-like object is required, not 'str'
```

The following diff fixes it for me, but may be breaks some other python versions.
```diff
--- a/urls-original.py
+++ b/urls.py
@@ -82,15 +82,15 @@ def path2url(path):
         # Make sure directory names have a trailing slash.
         # Otherwise relative URIs are resolved from the parent directory.
         path += os.path.sep
+    path = pathname2url(path)
     if isinstance(path, unicode):
         path = path.encode(FILESYSTEM_ENCODING)
-    path = pathname2url(path)
-    if path.startswith('///'):
+    if path.startswith(b'///'):
         # On Windows pathname2url(r'C:\foo') is apparently '///C:/foo'
         # That enough slashes already.
-        return 'file:' + path
+        return b'file:' + path
     else:
-        return 'file://' + path
+        return b'file://' + path
```

**Comment by liZe** [comment]
See #132.

**Comment by liZe** [comment]
Should be fixed now that #132 is fixed!
